### ISSUE TITLE
Failure to parse hex number

### ISSUE THREAD

**Issue by EricG-Personal** [uiri/toml#169]
According to https://github.com/toml-lang/toml#toml,

`var = 0x100`

is valid toml. However,

```
> import toml
> a="var = 0x100"
> toml.loads(a)
```

produces an error:

```
TomlDecodeError: Weirdness with leading zeroes or underscores in your number.
```

The code should produce the same result as:

```
> import toml
> a="var = 256"
> toml.loads(a)

{'var': 256}
```

**Comment by uiri** [comment]
Hi,

This is not a bug. From the [0.4.0 spec](https://github.com/toml-lang/toml/blob/master/versions/en/toml-v0.4.0.md#integer):

> Hex, octal, and binary forms are not allowed.

**Comment by Fieoner** [comment]
Hi Will,

According to the [0.5.0 spec](https://github.com/toml-lang/toml/blob/master/versions/en/toml-v0.5.0.md#user-content-integer) they are supposed to be allowed:

> Non-negative integer values may also be expressed in hexadecimal, octal, or binary.

**Comment by uiri** [comment]
Marking this as closed by 87b5e22 :)
